gammapy/irf/psf/map.py

```diff
@@ -16,6 +17,25 @@
 __all__ = ["PSFMap", "RecoPSFMap"]
 
 
+PSF_MAX_OVERSAMPLING = 4
+
+
+def _psf_upsampling_factor(psf, geom, position, energy=None, precision_factor=12):
+    if energy is None:
+        energy = geom.axes[psf.energy_name].center
+    psf_r68 = psf.containment_radius(
+        0.68, geom.axes[psf.energy_name].center, position=position
+    )
+    psf_r68 = np.percentile(psf_r68, 50)
+    base_factor = (2 * psf_r68 / geom.pixel_scales.max()).to_value("")
+    factor = np.minimum(
+        int(np.ceil(precision_factor / base_factor)), PSF_MAX_OVERSAMPLING
+    )
+    if isinstance(geom, HpxGeom):
+        factor = int(2 ** np.ceil(np.log(factor) / np.log(2)))
+    return factor
+
+
 class IRFLikePSF(PSF):
```

Comment:
Can you add a small docstring to this class?

Reply:
it is unrelated so it can be done in a sperate PR